packages/block-library/src/group/edit.js

```diff
@@ -124,6 +126,7 @@ function GroupEdit( { attributes, name, setAttributes, clientId } ) {
 			? blockProps
 			: { className: 'wp-block-group__inner-container' },
 		{
+			dropZoneElement: ref.current,
```

Comment:
So in #56312 a similar change was made in Cover so that dragging over the empty area inside the block would make the inner drop zone show, and that caused the drop zone between blocks with no margins to be hidden. But this change is now necessary to make it possible for the in between drop zone to show between Groups?

My main question is how this may affect third party containers - will all containers now need a `dropZoneElement` ref passed to `useInnerBlocksProps`?

Reply:
Good question!

> My main question is how this may affect third party containers - will all containers now need a dropZoneElement ref passed to useInnerBlocksProps?

With this PR (and right now) I think they might need to if they want to allow the threshold dragging behaviour in this PR. It's a tricky one, because we can't _really_ know exactly what a 3rd party block is doing with its UI and where its inner blocks will be placed in relation to anything else displayed within the block itself (a bit like how we have things like Media & Text, which is a non-standard kind of container block).

I think I'd lean slightly toward guarding the behaviour in this PR behind the `dropZoneElement` behaviour for now (and having it be manually passed to `useInnerBlocksProps`) while we're still fine-tuning how all the drag and drop behaviour works. Then, perhaps further down the track, we could look at making it all happen automatically within `useInnerBlocksProps` somehow? The tricky part in thinking about that, though, is that not all blocks might be using a `ref` attached to their outer wrapper, so it might be hard to make it all automatic. Still, worth thinking about!

Reply:
> we can't _really_ know exactly what a 3rd party block is doing with its UI and where its inner blocks will be placed in relation to anything else

Yeah this is a good point. In testing with plain WP 6.4 I notice that it's also not possible to drop between two Columns blocks, but it is between Media & Text blocks. Those behaviours are unaltered in this PR so I guess that means regressions are unlikely? 

I'm still not sure I quite grasp why the same thing that made dropping between blocks stop working for Cover in #56312 (passing the ref to `useInnerBlocksProps`) is making dropping between Group blocks work in this PR 😅

Reply:
It's already broken though - I guess the difference is that on trunk a drop indicator appears, apparently between the blocks, but in reality dropping over it just adds the dropped block to one of the Groups.

This happens in WP 6.4 with or without Gutenberg:


https://github.com/WordPress/gutenberg/assets/8096000/9b4442fb-1da8-4fd6-8db1-1eeccbb238d3

Reply:
Gotcha! Yep, on trunk, I think you're right, it looks like it's just dropping within 👍

I must admit, I do find it hard to maintain a good mental model for how all this hooks together 😄

Thanks again for giving this a look!